src/Workspaces/Core/MSBuild/Rpc/RpcClient.cs

```diff
@@ -139,14 +139,36 @@ public Task InvokeAsync(int targetObject, string methodName, List<object?> param
 
         var requestJson = JsonConvert.SerializeObject(request, JsonSettings.SingleLineSerializerSettings);
 
-        using (await _sendingStreamSemaphore.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
+        try
         {
-            await _sendingStream.WriteLineAsync(requestJson).ConfigureAwait(false);
-#pragma warning disable CA2016 // https://github.com/dotnet/roslyn/issues/71580
-            await _sendingStream.FlushAsync().ConfigureAwait(false);
-#pragma warning restore CA2016 
+            // The only cancellation we support is cancelling before we are able to write the request to the stream; once it's been written
+            // the other side will execute it to completion. Thus cancellationToken is checked here, but nowhere else.
```

Comment:
And I should say: it's absolutely possible to be fancier here with cancellation handling, but the whole point of this RPC system is to be trivial. And practically MSBuildWorkspace often can't pass a cancellation token _anyways_ so spending time on making this fancier is just a waste of time.